frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow.tsx

```diff
@@ -0,0 +1,315 @@
+  DashboardRefreshPeriodControls,
+  EmbedThemeControls,
+} from "metabase/dashboard/types";
+import { useSelector } from "metabase/lib/redux";
+import { getPulseFormInput } from "metabase/pulse/selectors";
+import {
+  canManageSubscriptions as canManageSubscriptionsSelector,
+  getUserIsAdmin,
+} from "metabase/selectors/user";
+import type { Collection, Dashboard } from "metabase-types/api";
+
+import { DashboardBookmark } from "../DashboardBookmark";
+import { ExtraEditButtonsMenu } from "../ExtraEditButtonsMenu/ExtraEditButtonsMenu";
+
+import { DashboardHeaderActionDivider } from "./DashboardHeader.styled";
+import {
+  AddActionElementButton,
+  AddFilterParameterButton,
+  AddHeadingOrTextButton,
+  AddLinkCardButton,
+  AddQuestionButton,
+  AddSectionButton,
+  AddTemporalUnitButton,
+  CopyAnalyticsDashboardButton,
+  DashboardActionMenu,
+  DashboardInfoButton,
+  DashboardSubscriptionButton,
+  EditDashboardButton,
+  FullscreenAnalyticsDashboard,
+  getExtraButtons,
+  shouldRenderSubscriptionButton,
+  FullscreenToggle,
+  NightModeToggleButton,
+} from "./buttons";
+
+export const BUTTON_CONFIG = {
+  ADD_QUESTION: "ADD_QUESTION",
+  ADD_HEADING_OR_TEXT: "ADD_HEADING_OR_TEXT",
+  ADD_LINK_CARD: "ADD_LINK_CARD",
+  ADD_SECTION: "ADD_SECTION",
+  ADD_TEMPORAL_UNIT: "ADD_TEMPORAL_UNIT",
+  ADD_FILTER_PARAMETER: "ADD_FILTER_PARAMETER",
+  ADD_ACTION_ELEMENT: "ADD_ACTION_ELEMENT",
+  EXTRA_EDIT_BUTTONS_MENU: "EXTRA_EDIT_BUTTONS_MENU",
+  COPY_ANALYTICS_DASHBOARD: "COPY_ANALYTICS_DASHBOARD",
+  EDIT_DASHBOARD: "EDIT_DASHBOARD",
+  DASHBOARD_SUBSCRIPTION: "DASHBOARD_SUBSCRIPTION",
+  DASHBOARD_EMBED_ACTION: "DASHBOARD_EMBED_ACTION",
+  REFRESH_WIDGET: "REFRESH_WIDGET",
+  NIGHT_MODE_TOGGLE: "NIGHT_MODE_TOGGLE",
+  FULLSCREEN_TOGGLE: "FULLSCREEN_TOGGLE",
+  DASHBOARD_HEADER_ACTION_DIVIDER: "DASHBOARD_HEADER_ACTION_DIVIDER",
+  DASHBOARD_BOOKMARK: "DASHBOARD_BOOKMARK",
+  DASHBOARD_INFO: "DASHBOARD_INFO",
+  DASHBOARD_ACTION_MENU: "DASHBOARD_ACTION_MENU",
+  FULLSCREEN_ANALYTICS_DASHBOARD: "FULLSCREEN_ANALYTICS_DASHBOARD",
+} as const;
+
+type ButtonKey = keyof typeof BUTTON_CONFIG;
+
+type DashboardHeaderButtonRowProps = WithRouterProps & {
+  collection?: Collection;
+  isEmpty?: boolean;
+  isPublic?: boolean;
+} & DashboardRefreshPeriodControls &
+  DashboardFullscreenControls &
+  Pick<
+    EmbedThemeControls,
+    "hasNightModeToggle" | "isNightMode" | "onNightModeChange"
+  >;
+
+type HeaderButtonProps = {
+  isEditing: boolean;
+  canEdit: boolean;
+  hasModelActionsEnabled: boolean;
+  isAnalyticsDashboard: boolean;
+  dashboard: Dashboard;
+  canManageSubscriptions: boolean;
+  formInput: any;
+  isAdmin: boolean;
+} & DashboardHeaderButtonRowProps;
+
+type ButtonConfig = {
+  component: (props: HeaderButtonProps) => ReactNode;
+  enabled: (props: HeaderButtonProps) => boolean;
+};
+
+const buttonConfigs: Record<ButtonKey, ButtonConfig> = {
+  [BUTTON_CONFIG.ADD_QUESTION]: {
+    component: () => <AddQuestionButton />,
```

Comment:
you should keep `key` like we did previously, since this raise `key` error in console.
![image](https://github.com/user-attachments/assets/26d52a83-3f57-414e-981b-80d32edf2fae)

e.g.
https://github.com/metabase/metabase/blob/515af74d44bd4f95c2e4d68c556f49fe07796ad7/frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.tsx#L182-L188